query/Updater.java

```diff
@@ -71,8 +71,10 @@ public static Updater create(Reasoner reasoner, ConceptManager conceptMgr, TypeQ
             insertRegistry.variables().forEach(Inserter::validate);
 
             assert query.match().namedVariablesUnbound().containsAll(query.namedDeleteVariablesUnbound());
-            HashSet<UnboundVariable> filter = new HashSet<>(query.namedDeleteVariablesUnbound());
-            filter.addAll(query.namedInsertVariablesUnbound());
+
+            HashSet<UnboundVariable> filter = new HashSet<>(query.match().namedVariablesUnbound());
```

Comment:
Try not to add new lines if they're not needed (so they mean something when we use them where they are needed)